contracts/types/PoolId.sol

```diff
@@ -7,7 +7,9 @@ type PoolId is bytes32;
 
 /// @notice Library for computing the ID of a pool
 library PoolIdLibrary {
-    function toId(PoolKey memory poolKey) internal pure returns (PoolId) {
-        return PoolId.wrap(keccak256(abi.encode(poolKey)));
+    function toId(PoolKey memory poolKey) internal pure returns (PoolId poolId) {
+        assembly("memory-safe") {
```

Comment:
also maybe just stick with `/// @solidity memory-safe-assembly` convention for now to be consistent with the rest of the repo. we might change these later...

https://github.com/Uniswap/v4-core/blob/e112589c1eb264e74e8626f6b91fbf4d8ba54639/contracts/types/BalanceDelta.sol#L10